client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,217 @@
+  import { lobbyReturn } from "$lib/firebase/firebase-functions";
+  import { authStore as user } from "$stores/auth";
+
+  export let lobbyCode: string;
+  export let lobby: Lobby;
+  export let privatePlayer: PrivatePlayer;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishList: string = "";
+  let catfishes: string[];
+  let isHost: boolean;
+  onMount(() => {
+    // check if the user is the host
+    if ($user !== null) {
+      if (lobby.uids.indexOf($user.uid) == 0) {
+        isHost = true;
+      } else {
+        isHost = false;
+      }
+    }
+    // format a string list to have commas and "and" when needed
+    const formatter = new Intl.ListFormat("en", { style: "long", type: "conjunction" });
+    // get the list of catfishes from lobby.ts and put them into a variable
+    catfishes = lobby.players
+      .filter((p) => {
+        return p.role == "CATFISH";
+      })
+      .map((c) => c.displayName);
+    catfishList = formatter.format(catfishes);
+
+    // grab the current user's role
+    const myRole = privatePlayer.role;
+    // update end variable depending on the winner type and the current player's role
+    if (myRole == "CAT") {
+      if (lobby.winner == "CAT") {
+        end = "Cat Win";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Cat Lose";
+      }
+    } else {
+      if (lobby.winner == "CAT") {
+        end = "Catfish Lose";
+      } else if (lobby.winner == "CATFISH") {
+        end = "Catfish Win";
+      }
+    }
+  });
+</script>
+
+<!-- TODO: Get different images for each screen -->
+<!-- Scenario 1 -->
+<div class="container">
+  <!-- Check the user's role and display the correct win or loss page accordingly -->
+  {#if end == "Cat Win"}
+    <div class="banner">
+      <h2>Hooray!</h2>
+      <h2>You have sniffed out the cat fish!</h2>
+    </div>
+    <div class="image">
+      <img src="/images/winloss/HappyCat.png" alt="happy cat" />
+    </div>
+    <div class="displayname">
+      <!-- Check the number of catfish for grammar purposes -->
+      {#if catfishes.length > 1}
+        <h2>{catfishList} were sentenced to 9 lives in purrison!</h2>
+      {:else}
+        <h2>{catfishList} was sentenced to 9 lives in purrison!</h2>
+      {/if}
+      <!-- If the player is the host, show the return button -->
+      {#if isHost}
+        <button
+          on:click={() => {
+            lobbyReturn({ code: lobbyCode });
+          }}>Return to Lobby</button
+        >
+      {:else}
+        <!-- Otherwise show a message -->
+        <span class="button">Waiting for host to return to lobby...</span>
+      {/if}
+    </div>
+    <!-- Scenario 2 -->
+  {:else if end == "Cat Lose"}
+    <div class="banner">
+      <h2>You have cat to be kitten me!</h2>
+      {#if catfishes.length > 1}
+        <h2>The impawster, {catfishList} was not caught!</h2>
+      {:else}
+        <h2>The impawsters, {catfishList} were not caught!</h2>
+      {/if}
```

Comment:
if catfishes.length is equal to 1 it will display `The impawsters, {catfishList} were not caught` because 1 isnt greater than 1